### PR TITLE
feat: prebuild minidump

### LINKED ISSUE #41: Prebuild node-minidump
We can prebuild minidump executables and ship them in the npm package. The build requirements on MacOS (Xcodebuild) seems out of reach for some of the users.

Addresses https://github.com/atom/atom/pull/21792#issuecomment-771090959

### PR DESCRIPTION
**Pull request by aminya**
This makes it possible to include the binaries in the npm package. 

### Description

The build script makes a `bin` folder that includes the binaries. These binaries are used by the main file. 
```
bin
 |_linux-x64
	|_dump_syms
	|_minidump_dump
	|_minidump_stackwalk
 |_darwin-x64
	|_dump_syms
	|_minidump_dump
	|_minidump_stackwalk
```

I assumed that the release process is done after manually by downloading the artifacts from the last GitHub action run. I can add an automatic release action if you prefer this to be done by the semantic-release bot.

### Issues
Fixes #41 

### Verification
A successful CI run:
https://github.com/aminya/node-minidump/actions/runs/528963738

The artifacts: 
https://github.com/aminya/node-minidump/suites/1945608672/artifacts/38471951

### CODE REVIEW

**Comment by aminya** [comment]
@nornagon Could you review this?

File: package.json [hunk 1]
```diff
@@ -34,6 +34,7 @@
     "/lib",
     "index.d.ts",
     "build.js",
-    "deps"
+    "deps",
+    "bin"
```
Comment (nornagon) [inline package.json:38]: Does this include binaries for all OSes in every release package? I'm not sure that's the right way to go about prebuilding. I think https://www.npmjs.com/package/prebuild-install is the de facto standard way to do this sort of thing.

**Review by nornagon — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (aminya) [reply package.json:38]: This PR takes the prebuildify approach.
https://github.com/prebuild/prebuildify

Quoting the readme:

> Always use prebuildify --@mafintosh

The difference with those tools is that those are made for `.node` files, but the executables in this package are not `.node` files. They are real exe files.

Comment (nornagon) [reply package.json:38]: hm, alright, i'm convinced on the approach then.

two more qs:
1. This doesn't appear to build on Windows in CI—will this automatically build the binaries the old way if they're not present for the platform/arch?
2. What's the expected process for forming an npm release? Does it involve manually downloading the binaries from CI? If so I think that's something we should aim to automate, or at the very least document.

Comment (aminya) [reply package.json:38]: > 1. This doesn't appear to build on Windows in CI—will this automatically build the binaries the old way if they're not present for the platform/arch?

Yes. Any other platform or arch will run the build script fully.  I was planning to add Windows support. After we merged this, we can look into that.

> 2. What's the expected process for forming an npm release? Does it involve manually downloading the binaries from CI? If so I think that's something we should aim to automate, or at the very least document.

Currently, the artifacts of the latest GitHub action run should be downloaded and placed under `bin` folder. 

If you want to make this automatic, we should use semantic-release bot. We need a `NPM_TOKEN` for that. The GitHub actions can download the artifacts automatically and release a new npm version.

Comment (nornagon) [reply package.json:38]: For now let's just add a note in the README about how to form a release.

Comment (aminya) [reply package.json:38]: Added the instructions in 
https://github.com/electron/node-minidump/pull/42/commits/0f360d78e4dfc15e3a12390ac05074a10c4c2a12

**Review by nornagon — APPROVED** [review]

**Comment by nornagon** [comment]
Hey there, while publishing this I noticed that the Linux binaries are quite large, around 15 MB. I think this is because they are unstripped. Running `strip` on the binaries reduces the size to about 1.5 MB.

Can we find a way to strip the linux binaries to reduce the download size? The breakpad source is itself already quite large.

**Comment by aminya** [comment]
> Hey there, while publishing this I noticed that the Linux binaries are quite large, around 15 MB. I think this is because they are unstripped. Running `strip` on the binaries reduces the size to about 1.5 MB.
> 
> Can we find a way to strip the linux binaries to reduce the download size? The breakpad source is itself already quite large.

We can do it for all the binaries. Here is the code:
https://www.github.com/prebuild/prebuild/tree/master/strip.js

**Pull request merged**